cmd/admin-handlers.go

```diff
@@ -979,6 +980,28 @@ func (adminAPI adminAPIHandlers) SetConfigHandler(w http.ResponseWriter, r *http
 		return
 	}
 
+	var config serverConfigV19
+	err = json.Unmarshal(configBytes, &config)
+
+	if err != nil {
+		errorIf(err, "Failed to unmarshal config from request body.")
+		writeErrorResponse(w, toAPIErrorCode(err), r.URL)
+		return
+	}
+
+	if accessKey := os.Getenv("MINIO_ACCESS_KEY"); accessKey != "" {
+		if accessKey != config.Credential.AccessKey {
+			writeErrorResponse(w, ErrAdminCredentialsMismatch, r.URL)
+			return
+		}
+	}
+	if secretKey := os.Getenv("MINIO_SECRET_KEY"); secretKey != "" {
+		if secretKey != config.Credential.SecretKey {
+			writeErrorResponse(w, ErrAdminCredentialsMismatch, r.URL)
+			return
+		}
+	}
```

Comment:
have a look at handleCommonEnvVars() where we set globalIsEnvCreds
so you can do
```
if globalIsEnvCreds {
    creds := serverConfig.GetCredential()
    if config.Credential.AccessKey != creds.AccessKey {...}
    if config.Credential.SecretKey != creds.SecretKey {...}
}
```

Reply:
that's a good suggestion.done